pkg/utils/apply/apply.go

```diff
@@ -397,3 +407,14 @@ func DryRunAll() ApplyOption {
 		return nil
 	}
 }
+
+// isUpdatableResource check whether the resource is updatable
+// Resource like v1.Service cannot unset the spec field (the ip spec is filled by service controller)
+func isUpdatableResource(desired client.Object) bool {
```

Comment:
I think I've written such kinds of code before, but I forget where.

Reply:
I think that part of code is removed. Let me find it in history archives.

Reply:
Are you referring to https://github.com/kubevela/kubevela/blob/8de36ff2950b3c757f912e81d2fc4be3ea4b60af/pkg/controller/core.oam.dev/v1alpha2/application/dispatch/dispatch.go#L285?

Reply:
Yes

Reply:
Just make sure no duplicate code

Reply:
No, that code only exists in release 1.1, not on master. Besides, they have different working mechanism. (Although use similar names)